error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -650,6 +651,42 @@ Flux<T> after(Flux<? extends Iterable<T>> flux, int prefetch) {
     }
   }
 
+  /**
+   * Prefer {@link Mono#doOnError(Class, Consumer)} over {@link Mono#doOnError(Predicate, Consumer)}
+   * where possible.
+   */
+  static final class MonoDoOnError<T> {
+    @BeforeTemplate
+    Mono<T> before(
+        Mono<T> mono, Class<? extends Throwable> clazz, Consumer<? super Throwable> onError) {
+      return mono.doOnError(clazz::isInstance, onError);
```

Comment:
Next to `clazz::isInstance` users may also write `t -> t instanceof clazz`. Rather than adding `Refaster.anyOf` in a bunch of places, let's instead add a `BugChecker` that maps the latter to the former. (For technical reasons IIUC we can't do this with Refaster.)

Reply:
I'll work on this on a separate PR 👌

Reply:
... and that PR is #323 :)